protos/protos/public/modeldb/versioning/Dataset.proto

```diff
@@ -56,6 +56,15 @@ message PathDatasetDiff {
 
 message S3DatasetComponentDiff {
     PathDatasetComponentDiff path = 1;
+    S3VersionIdDiff s3_version_id = 2;
+    DiffStatusEnum.DiffStatus status = 3;
```

Comment:
@conradoverta : i am finding it difficult to merge S3DatasetComponent without the status here. Does it make sense to have a DiffStatus here?

Reply:
The recommendation is to add a `S3DatasetComponentVersionDiff` that has status, A, B and C, like we do for everything else like command line in the environment. This should automatically fall into place with the autogen code.